src/addons/Portal/Portal.js

```diff
@@ -0,0 +1,255 @@
+import React, { Children, cloneElement, PropTypes } from 'react'
+import ReactDOM from 'react-dom'
+
+import {
+  AutoControlledComponent as Component,
+  // customPropTypes,
+  // getElementType,
+  // getUnhandledProps,
+  keyboardKey,
+  makeDebugger,
+  META,
+} from '../../lib'
+
+const debug = makeDebugger('portal')
+
+const _meta = {
+  name: 'Portal',
+  type: META.TYPES.ADDON,
+}
+
+/**
+ * A component that allows you to render children outside their parent.
+ * @see Modal
+ */
+class Portal extends Component {
+  static propTypes = {
+    /** Primary content */
+    children: PropTypes.node.isRequired,
+
+    /** Classes to be added to the portal node element. */
+    className: PropTypes.string,
+
+    /** Controls whether or not the portal should close on a click outside. */
+    closeOnDocumentClick: PropTypes.bool,
+
+    /** Controls whether or not the portal should close when escape is pressed is displayed. */
+    closeOnEscape: PropTypes.bool,
+
+    /** Controls whether or not the portal should close on blur of the trigger. */
+    closeOnTriggerBlur: PropTypes.bool,
+
+    /** Controls whether or not the portal should close when mousing out of the trigger. */
+    closeOnTriggerMouseLeave: PropTypes.bool,
+
+    /** Initial value of open. */
+    defaultOpen: PropTypes.bool,
+
+    /** Called when open changes from true to false */
+    onClose: PropTypes.func,
+
+    /** Called when open changes from false to true */
+    onOpen: PropTypes.func,
+
+    /** Controls whether or not the portal is displayed. */
+    open: PropTypes.bool,
+
+    /** Controls whether or not the portal should open when the trigger is clicked. */
+    openOnTriggerClick: PropTypes.bool,
+
+    /** Controls whether or not the portal should open on focus of the trigger. */
+    openOnTriggerFocus: PropTypes.bool,
+
+    /** Controls whether or not the portal should open when mousing over the trigger. */
+    openOnTriggerMouseOver: PropTypes.bool,
+
+    /** Element to be rendered in-place where the portal is defined. */
+    trigger: PropTypes.node,
+  }
+
+  static defaultProps = {
+    closeOnEscape: true,
+    closeOnDocumentClick: true,
+    openOnTriggerClick: true,
+  }
+
+  static autoControlledProps = [
+    'open',
+  ]
+
+  static _meta = _meta
+
+  componentDidMount() {
+    if (this.state.open) {
+      this.renderPortal()
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) { // eslint-disable-line complexity
```

Comment:
Very few branches in this method, is `// eslint-disable-line complexity` still needed?